cirq-core/cirq/devices/device.py

```diff
@@ -178,3 +179,83 @@ def __iter__(self) -> Iterator['cirq.Qid']:
 
     def __contains__(self, item: 'cirq.Qid') -> bool:
         return item in self.qids
+
+
+@value.value_equality
+class DeviceMetaData:
+    """Parent type for all device specific metadata classes."""
+
+    def qubit_set(self) -> Optional[FrozenSet['cirq.Qid']]:
+        """Returns a set of qubits on the device, if possible.
+
+        Returns:
+            Frozenset of qubits on device if specified, otherwise None.
+        """
+        return cast(Optional[FrozenSet['cirq.Qid']], self._qubits_set)
```

Comment:
`cast` shouldn't be necessary here - you can define the type of `self._qubits_set` in the constructor.

Reply:
Changed.